flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -253,6 +262,35 @@ public String getShardIterator(KinesisStreamShard shard, String shardIteratorTyp
 		return getShardIteratorResult.getShardIterator();
 	}
 
+	/**
+	 * Determines whether the exception can be recovered from using
+	 * exponential-backoff
+	 * 
+	 * @param ex
+	 *            Exception to inspect
+	 * @return <code>true</code> if the exception can be recovered from, else
+	 *         <code>false</code>
+	 */
+	protected static boolean isRecoverableException(AmazonServiceException ex) {
+		if (ex.getErrorType() == null) {
+			return false;
+		}
+
+		switch (ex.getErrorType()) {
+		case Client:
+			if (ex instanceof ProvisionedThroughputExceededException) {
```

Comment:
It'll probably be cleaner to just do `ex instanceof ProvisionedThroughputExceededException `